packages/strapi-plugin-upload/admin/src/containers/HomePage/reducer.js

```diff
@@ -2,34 +2,49 @@ import { fromJS } from 'immutable';
 
 const initialState = fromJS({
   data: [],
+  dataCount: 0,
   dataToDelete: [],
 });
 
 const reducer = (state, action) => {
   switch (action.type) {
+    case 'CLEAR_DATA_TO_DELETE':
+      return state.update('dataToDelete', () => fromJS([]));
     case 'GET_DATA_SUCCEEDED':
       return state.update('data', () => fromJS(action.data));
+    case 'GET_DATA_COUNT_SUCCEEDED':
+      return state.update('dataCount', () => fromJS(action.count));
     case 'ON_CHANGE_DATA_TO_DELETE': {
       const { value, id } = action;
 
       if (value) {
+        const item = state.get('data').find(item => item.get('id') === id);
```

Comment:
You are getting the id directly from your action so this is not needed

Reply:
Do you mean i should directly use the whole item instead of its id?

Reply:
You can just use an array of ids instead of an array of files so you won't need all this logic.

Reply:
I don't want `dataToDelete` to be an array of ids because we use an array of object to select the items in the modal list. We have to do it this way because of the pagination. We can select items though pagination but we also have to use all of them for the select tab display.

The list, which is the same component in the modal and the listView, now use an array of object and not an array of ids. I would prefer to be consistent and to let it as an array of objects to avoid multiple management cases inside a generic compo.